### DISCUSSION TITLE
How to show a search icon in the navigation bar?

### DISCUSSION THREAD

**Discussion by l4ndfr05ch** [datenstrom/community#951] [Ask a question] [answered]
Hello,

I would like to change the header and the navigation bar. The home button (red circle in the image) should be replaced by the magnifying glass symbol (search function). In which file can I change this cleanly and permanently? Unfortunately, I have not found a suitable entry in the corresponding files. 
I use the Stockholm theme.

Best regards,
Marko
![menu](https://github.com/datenstrom/yellow/assets/162504154/8ee2592c-2e33-49ca-925c-929386b4dc4e)

**Comment by l4ndfr05ch** [comment]
Now I have found a solution for myself. I don't quite like it, but it fulfils its purpose. I have simply added a new menu item to the navigation bar and added the entry [search /search/] to page.md. It works.
Perhaps someone has a more elegant solution? For example, I would like to have the search symbol in the menu. But I don't know in which file I have to add which entry exactly. 

my website: [https://www.landfrosch.com](https://www.landfrosch.com)

**Comment by schulle4u** [comment]
In your `page.md` in the `content/search` folder you can set `TitleNavigation` to change what should be displayed in the site's menu, e.g.: 

```
--- 
Title: Suche
TitleNavigation: 🔎
---
```

I don't know if this also works with fontawesome symbols, maybe its better then to put a search link into a separate navigation bar - not only for accessibility reasons. For an example how to use a sidebar navigation, have a look at [Paris theme](https://github.com/annaesvensson/yellow-paris). If I remember this theme correctly you can create a sidebar by simply adding a `sidebar.md` to the `content/shared` folder with your desired menu entries.

**Comment by markseuffert** [comment, marked as answer]
You could make your own layout file with HTML. Here's an example file `system/layouts/navigation.html` with search icon.

```
<?php $pages = $this->yellow->content->top() ?>
<?php $this->yellow->page->setLastModified($pages->getModified()) ?>
<div class="navigation" role="navigation" aria-label="<?php echo $this->yellow->language->getTextHtml("coreNavigation") ?>">
<ul>
<?php foreach ($pages as $page): ?>
<li><a<?php echo $page->isActive() ? " class=\"active\" aria-current=\"page\"" : "" ?> href="<?php echo $page->getLocation(true) ?>"><?php echo $page->getHtml("titleNavigation") ?></a></li>
<?php endforeach ?>
<li><a<?php echo $this->yellow->page->location=="/search/" ? " class=\"active\" aria-current=\"page\"" : "" ?> href="<?php echo $this->yellow->page->getBase(true)."/search/" ?>"><i class="icon icon-search" aria-label="Search"></i></a></li>
</ul>
</div>
<div class="navigation-banner"></div>
```

See also https://github.com/annaesvensson/yellow-icon

**Reply by l4ndfr05ch** [reply]
Thank you very, very much! I didn't expect so much support. I'm very surprised and delighted. Unfortunately, my programming skills are not sufficient to write this code. I have replaced your code with the original code. Everything works perfectly .
I've only been working with Yellow for about a week and I'm really impressed. I particularly like the very minimalistic design. That's all I need.
So thanks again for your support!!!

**Reply by GiovanniSalmeri** [reply]
Just two little tips from my part:
1. Many years ago I learned to program in Perl (and later in PHP) by figuring out and modifying pre-existing programs: small changes at first, then gradually more substantial changes, eventually even programs written from scratch. In this way you learn almost without realizing it.
2. One of the most useful features of Yellow is the fact that you can install it very easily on your computer (you just need to install PHP first). This allows you to test very quickly, without messing with your online site. If you haven't done it yet, I recommend you try it out!
